### PR TITLE
KAA-932 Client bug fixes

### PR DESCRIPTION
**Pull request by Dubland**
/cc @rasendubi @forGGe

### CODE REVIEW

File: client/client-multi/client-c/src/kaa/platform-impl/common/kaa_tcp_channel.c [hunk 1]
```diff
@@ -202,10 +203,8 @@ kaa_error_t kaa_tcp_channel_create(kaa_transport_channel_interface_t *self
     }
 
     kaa_tcp_channel->supported_service_count = supported_service_count;
-    i = 0;
-    for (; i < supported_service_count; ++i) {
-        kaa_tcp_channel->supported_services[i] = supported_services[i];
-    }
+
+	memcpy(kaa_tcp_channel->supported_services, supported_services, sizeof(kaa_service_t) * supported_service_count);
```
Comment (forGGe) [inline client/client-multi/client-c/src/kaa/platform-impl/common/kaa_tcp_channel.c]: Convert tabs to spaces

File: client/client-multi/client-c/src/kaa/kaa_protocols/kaa_tcp/kaatcp_request.c [hunk 2]
```diff
@@ -285,11 +285,13 @@ kaatcp_error_t kaatcp_get_request_kaasync(const kaatcp_kaasync_t *message, char
     KAA_RETURN_IF_ERR(rval);
 
     if (message->sync_request) {
-        memcpy(cursor, message->sync_request, message->sync_request_size);
-        cursor += message->sync_request_size;
+        if (cursor + message->sync_request_size <= buf + first_buf_size) {
+	  memcpy(cursor, message->sync_request, message->sync_request_size);
```
Comment (forGGe) [inline client/client-multi/client-c/src/kaa/kaa_protocols/kaa_tcp/kaatcp_request.c]: Mixed indent

File: client/client-multi/client-c/test/kaatcp/kaatcp_request_test.c [hunk 3]
```diff
@@ -140,6 +140,23 @@ void test_kaatcp_ping(void)
...
+    KAA_TRACE_IN(logger);
+    kaatcp_kaasync_t kaa_sync_message;
+    kaa_sync_message.sync_request_size = 6;
+    kaa_sync_message.sync_request = "hello";
+    unsigned char buffer[100] = "";
+    size_t buffer_size = 5;
+    memset(buffer, 0xEA, 100);
+    ASSERT_EQUAL(kaatcp_get_request_kaasync(&kaa_sync_message, buffer, &buffer_size), KAATCP_ERR_BUFFER_NOT_ENOUGH);
+    for (int i = 10; i < 100; i++) {
+      ASSERT_EQUAL(buffer[i], 0xEA);
```
Comment (forGGe) [inline client/client-multi/client-c/test/kaatcp/kaatcp_request_test.c]: Fix indent

**Comment by forGGe** [comment]
Reviewed

**Comment by rasendubi** [comment]
Removing test is not a fix.

**Comment by forGGe** [comment]
Looks good.

**Comment by rasendubi** [comment]
LGTM
@forGGe ?

File: client/client-multi/client-c/test/kaatcp/kaatcp_request_test.c [hunk 4]
```diff
@@ -143,15 +143,19 @@ void test_kaatcp_ping(void)
...
-    memset(buffer, 0xEA, 100);
-    ASSERT_EQUAL(kaatcp_get_request_kaasync(&kaa_sync_message, buffer, &buffer_size), KAATCP_ERR_BUFFER_NOT_ENOUGH);
-    for (int i = 10; i < 100; i++) {
-        ASSERT_EQUAL(buffer[i], 0xEA);
+    kaatcp_kaasync_t kaasync;
+    char *payload = "payload";
+    unsigned char kaasync_buf[100] = "";
+    size_t kaasync_buf_size = 5;
+
+    kaatcp_fill_kaasync_message(payload, strlen(payload), 5, 0, 1, &kaasync);
```
Comment (forGGe) [inline client/client-multi/client-c/test/kaatcp/kaatcp_request_test.c]: magic numbers

File: client/client-multi/client-c/test/kaatcp/kaatcp_request_test.c [hunk 5]
```diff
@@ -143,15 +143,19 @@ void test_kaatcp_ping(void)
...
+    memset(kaasync_buf, 0xEA, 100);
```
Comment (forGGe) [inline client/client-multi/client-c/test/kaatcp/kaatcp_request_test.c]: use sizeof() operator here

File: client/client-multi/client-c/test/kaatcp/kaatcp_request_test.c [hunk 6]
```diff
@@ -143,15 +143,19 @@ void test_kaatcp_ping(void)
...
+
+    kaatcp_error_t rval = kaatcp_get_request_kaasync(&kaasync, kaasync_buf, &kaasync_buf_size);
+    ASSERT_EQUAL(rval, KAATCP_ERR_BUFFER_NOT_ENOUGH);
+
+    for (int i = 8; i < 100; i++) {
```
Comment (forGGe) [inline client/client-multi/client-c/test/kaatcp/kaatcp_request_test.c]: same

**Comment by forGGe** [comment]
Reviewed

**Comment by forGGe** [comment]
Looks good to me

**Pull request merged**